src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -167,158 +171,153 @@ public void Initialize()
 
             this.indexingTask = Task.Run(async () => await this.IndexAddressesContinuouslyAsync().ConfigureAwait(false));
 
+            this.asyncProvider.RegisterTask($"{nameof(AddressIndexer)}.{nameof(this.indexingTask)}", this.indexingTask);
+
             this.nodeStats.RegisterStats(this.AddInlineStats, StatsType.Inline, 400);
         }
 
         private async Task IndexAddressesContinuouslyAsync()
         {
-            try
-            {
-                Stopwatch watch = Stopwatch.StartNew();
+            Stopwatch watch = Stopwatch.StartNew();
 
-                while (!this.cancellation.IsCancellationRequested)
+            while (!this.cancellation.IsCancellationRequested)
+            {
+                if (DateTime.Now - this.lastFlushTime > this.flushChangesInterval)
                 {
-                    if (DateTime.Now - this.lastFlushTime > this.flushChangesInterval)
+                    this.logger.LogDebug("Flushing changes.");
+
+                    lock (this.lockObject)
                     {
-                        this.logger.LogDebug("Flushing changes.");
+                        this.addressIndexRepository.SaveAllItems();
+                        this.outpointsRepository.SaveAllItems();
+                        this.tipDataStore.Update(this.tipData);
+                    }
 
-                        lock (this.lockObject)
-                        {
-                            this.addressIndexRepository.SaveAllItems();
-                            this.outpointsRepository.SaveAllItems();
-                            this.tipDataStore.Update(this.tipData);
-                        }
+                    this.lastFlushTime = DateTime.Now;
 
-                        this.lastFlushTime = DateTime.Now;
+                    this.logger.LogDebug("Flush completed.");
+                }
 
-                        this.logger.LogDebug("Flush completed.");
-                    }
+                if (this.cancellation.IsCancellationRequested)
```

Comment:
Any particular reason this is being checked again here? At worst we execute the remainder of the loop before exiting.

Reply:
this is not anything I introduced in this PR, is just git that gets confused because I removed the wrapping try/catch